java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java

```diff
@@ -79,6 +81,20 @@ public static Schema deserialize(ByteBuffer buffer) {
     return convertSchema(org.apache.arrow.flatbuf.Schema.getRootAsSchema(buffer));
   }
 
+  /** Returns the schema from the message.
+   *
+   * <p>The message must have the schema field set or an
+   * exception will be raised.
+   */
+  public static Schema deserializeFromMessage(MessageMetadataResult buffer) {
```

Comment:
This is duplicated from MessageSerializer right? I saw you removed `deserialzeFromMessage` for the other messages, can you do the same here?

Reply:
I think I copied it from flight originally, but removed.